.eslintrc.json

```diff
@@ -17,6 +17,16 @@
                 "MemberExpression": 2,
                 "ignoredNodes": [ "JSXAttribute" ]
             }],
+        "import/order": ["error",
+            {
+                "alphabetize": { "order": "asc" },
+                "groups": ["builtin", "external", "internal", "parent", "sibling"],
+                "newlines-between": "always",
+                "pathGroupsExcludedImportTypes": ["react"],
+                "pathGroups": [
+                    { "pattern": "react", "group": "builtin", "position": "before" }
```

Comment:
Is that substring match? It either doesn't seem to treat "react-dom/client" as matching here, or it's treated as a separate group.

Reply:
This is super complicated https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/order.md

https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/order.md#pathgroups-array-of-objects


This uses `minimatch` under the hood, I just allowed `react` and `react-dom` as `react.*` also matches patternfly imports @patternfly/react-core etc.

Reply:
So with `debug`
```
matchOne {
  this: Minimatch {
    options: { debug: true },
    set: [ [Array] ],
    pattern: 'react(-dom)?',
    regexp: null,
    negate: false,
    comment: false,
    empty: false,
    partial: false,
    globSet: [ 'react(-dom)?' ],
    debug: [Function: debug],
    globParts: [ [Array] ]
  },
  file: [ 'timeformat' ],
  pattern: [
    /^(?=.)react\(-dom\)[^/]$/ {
      _glob: 'react(-dom)?',
      _src: '(?=.)react\\(-dom\\)[^/]'
    }
  ]
}
```

Reply:
Ah, don't sweat it -- I'd rather enforce this to be consistent, but "slightly weird", instead of the current wild mess. We can tweak this in a follow-up.